### PR TITLE
Ensure a program with top level statements is an executable.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #41704.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @RikkiGibson, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs [hunk 1]
```diff
@@ -1543,35 +1543,47 @@ internal bool DeclaresTheObjectClass
 
         internal EntryPoint? GetEntryPointAndDiagnostics(CancellationToken cancellationToken)
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs:1544]: Do we ever expect this function to return `null` any more now that we use `EntryPoint.None` to indicate "we looked for an entry point and didn't find one"?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs:1544]: > Do we ever expect this function to return null any more now that we use EntryPoint.None to indicate "we looked for an entry point and didn't find one"?

No, I'll remove the ``?``` from the return type. Good catch.

---
In reply to: [416818038](https://github.com/dotnet/roslyn/pull/43738#discussion_r416818038) [](ancestors = 416818038)

Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs:1544]: Consider also removing the null check on the return value in `MethodCompiler.GetEntryPoint`.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-infrastructure It looks like CI got stuck on this PR. There isn't even a link to manage/examine the jobs.

**Comment by jaredpar** [comment]
Not sure why the build isn't showing up in the checks. Here is the build link though that ran for this PR

- Unit Tests: https://dev.azure.com/dnceng/public/_build/results?buildId=622483
- Integration Tests: https://dev.azure.com/dnceng/public/_build/results?buildId=622484

Both succeeded. If you want me to merge based on that let me know and I'll hit the merge button (please specify squash, normal, etc ...)

**Pull request merged**